core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -42,7 +42,8 @@ import org.apache.spark.util.Utils
 
 abstract class AppBase(
     val eventLogInfo: Option[EventLogInfo],
-    val hadoopConf: Option[Configuration]) extends Logging
+    val hadoopConf: Option[Configuration],
+    val platform: Option[Platform] = None) extends Logging
```

Comment:
That's not what I exactly thought we are heading to. Creating that platform before processing the eventlog implies that we are not using the information/cluster-detection logic from the eventlog.